be/src/exec/vectorized/topn_node.cpp

```diff
@@ -219,7 +222,26 @@ pipeline::OpFactories TopNNode::decompose_to_pipeline(pipeline::PipelineBuilderC
 
     // step 0: construct pipeline end with sort operator.
     // get operators before sort operator
-    OpFactories operator_sink_with_sort = _children[0]->decompose_to_pipeline(context);
+    OpFactories pred_operators = _children[0]->decompose_to_pipeline(context);
+
+    // predecessor pipeline has multiple drivers that will produce multiple output streams, but sort operator is
+    // not parallelized now and can not accept multiple streams as input, so add a LocalExchange to gather multiple
+    // streams and produce one output stream piping into the sort operator.
+    DCHECK(!pred_operators.empty() && pred_operators[0]->is_source());
+    auto* source_operator = down_cast<SourceOperatorFactory*>(pred_operators[0].get());
+    if (source_operator->num_driver_instances() > 1) {
+        auto mem_mgr = std::make_shared<LocalExchangeMemoryManager>(config::vector_chunk_size);
+        auto local_exchange_source =
+                std::make_shared<LocalExchangeSourceOperatorFactory>(context->next_operator_id(), mem_mgr);
+        auto local_exchange = std::make_shared<PassthroughExchanger>(mem_mgr, local_exchange_source.get());
+        auto local_exchange_sink =
+                std::make_shared<LocalExchangeSinkOperatorFactory>(context->next_operator_id(), local_exchange);
+        pred_operators.emplace_back(std::move(local_exchange_sink));
+        context->add_pipeline(std::move(pred_operators));
+        // Multiple LocalChangeSinkOperators pipe into one LocalChangeSourceOperator.
+        local_exchange_source->set_num_driver_instances(1);
+        pred_operators.emplace_back(local_exchange_source);
```

Comment:
why use a moved obj?

Reply:
std::move indicate that the pred_operators is cleared, so it can be used again.

Reply:
I think the state of a moved object depend on it's implementation of move assign operator, generally, I suggest reset it before reuse it